File: sources/BlazorReduxDevToolsExtension/ReduxDevToolsExtension.cs
using System;
using Microsoft.AspNetCore.Blazor.Browser.Interop;

namespace BlazorReduxDevToolsExtension
{
    public class ReduxDevToolsExtension
    {
        public static event EventHandler<MessageReceivedEventArgs> MessageReceived;
        public static void OnMessageReceived(string message) => MessageReceived?.Invoke(null, new MessageReceivedEventArgs(message));
        public static bool IsAvailable()
        {
            return RegisteredFunction.Invoke<bool>("BlazorReduxDevToolsExtension.IsAvailable");
        }
        public static void Connect(ReduxDevToolsExtensionOptions options = null)
        {
            RegisteredFunction.Invoke<object>("BlazorReduxDevToolsExtension.Connect", options);
        }
        public static void Init(object state)
        {
            RegisteredFunction.Invoke<object>("BlazorReduxDevToolsExtension.Init", state);
        }
        public static void Subscribe()
        {
            RegisteredFunction.Invoke<object>("BlazorReduxDevToolsExtension.Subscribe");
        }
        public static void Send(object action, object state)
        {
            RegisteredFunction.Invoke<object>("BlazorReduxDevToolsExtension.Send", action, state);
        }
        public static void UnSubscribe()
        {
            RegisteredFunction.Invoke<object>("BlazorReduxDevToolsExtension.UnSubscribe");
        }
        public static void Disconnect()
        {
            RegisteredFunction.Invoke<object>("BlazorReduxDevToolsExtension.Disconnect");
        }
    }
}

File: sources/BlazorReduxDevToolsExtension/content/ReduxDevToolsExtension.js
const assemblyName = "BlazorReduxDevToolsExtension";
const namespace = "BlazorReduxDevToolsExtension";
const typeName = "ReduxDevToolsExtension";
const methodName = "OnMessageReceived";

const withDevTools = (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION__);
const BlazorReduxDevToolsExtension = {};
Blazor.registerFunction('BlazorReduxDevToolsExtension.IsAvailable', function () {
    return withDevTools !== undefined;
});
Blazor.registerFunction('BlazorReduxDevToolsExtension.Connect', function (options) {
    BlazorReduxDevToolsExtension.devTools = withDevTools.connect(options);
});
Blazor.registerFunction('BlazorReduxDevToolsExtension.Init', function (state, subscribe) {
    
    return BlazorReduxDevToolsExtension.devTools.init(state);
});
Blazor.registerFunction('BlazorReduxDevToolsExtension.Send', function (action, state) {
    return BlazorReduxDevToolsExtension.devTools.send(action, state);
});
Blazor.registerFunction('BlazorReduxDevToolsExtension.Subscribe', function () {
    const messageReceivedMethod = Blazor.platform.findMethod(
        assemblyName,
        namespace,
        typeName,
        methodName
    );

    BlazorReduxDevToolsExtension.subscription = BlazorReduxDevToolsExtension.devTools.subscribe((message) => {
        var json = JSON.stringify(message);
        Blazor.platform.callMethod(messageReceivedMethod, null, [Blazor.platform.toDotNetString(json)]);
    });
});
Blazor.registerFunction('BlazorReduxDevToolsExtension.UnSubscribe', function () {
    BlazorReduxDevToolsExtension.devTools.unsubscribe();
});
Blazor.registerFunction('BlazorReduxDevToolsExtension.Disconnect', function () {
    withDevTools.disconnect();
});